scripts/tests/twister/test_config_parser.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Tests for config_parser.py
+"""
+import os
+import pytest
+import mock
+import scl
+
+from twisterlib.config_parser import TwisterConfigParser, extract_fields_from_arg_list, ConfigurationError
+
+
+def test_extract_single_field_from_string_argument():
+    target_fields = {"FIELD1"}
+    arg_list = "FIELD1=value1 FIELD2=value2 FIELD3=value3"
+    extracted_fields, other_fields = extract_fields_from_arg_list(
+        target_fields, arg_list)
+
+    assert extracted_fields == {"FIELD1": ["value1"]}
+    assert other_fields == "FIELD2=value2 FIELD3=value3"
+
+
+def test_no_fields_to_extract():
+    target_fields = set()
+    arg_list = "arg1 arg2 arg3"
+
+    extracted_fields, other_fields = extract_fields_from_arg_list(
+        target_fields, arg_list)
+
+    assert extracted_fields == {}
+    assert other_fields == "arg1 arg2 arg3"
+
+
+def test_missing_fields():
+    target_fields = {"CONF_FILE", "OVERLAY_CONFIG", "DTC_OVERLAY_FILE"}
+    arg_list = "arg1 arg2 arg3"
+    extracted_fields, other_fields = extract_fields_from_arg_list(
+        target_fields, arg_list)
+
+    assert extracted_fields == {"CONF_FILE": [], "OVERLAY_CONFIG": [], "DTC_OVERLAY_FILE": []}
+    assert other_fields == "arg1 arg2 arg3"
+
+def test_load_yaml_with_extra_args_and_retrieve_scenario_data(zephyr_base):
+    filename = "test_data.yaml"
+
+    yaml_data = '''
+    tests:
+      scenario1:
+        tags: ['tag1', 'tag2']
+        extra_args: '--CONF_FILE=file1.conf --OVERLAY_CONFIG=config1.conf'
+        filter: 'filter1'
+    common:
+      filter: 'filter2'
+    '''
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas','twister', 'testsuite-schema.yaml')
+    )
+
+    with mock.patch('builtins.open', mock.mock_open(read_data=yaml_data)):
+        parser = TwisterConfigParser(filename, loaded_schema)
+        parser.load()
+
+    scenario_data = parser.get_scenario('scenario1')
+    scenario_common = parser.common
+
+    assert scenario_data['tags'] == {'tag1', 'tag2'}
+    assert scenario_data['extra_args'] == ['--CONF_FILE=file1.conf', '--OVERLAY_CONFIG=config1.conf']
```

Comment:
Line too long.

Reply:
i don't get it #2 :)

Reply:
as before - 2 chars - I'm keeping it as it is:)